### PR TITLE
Use GitHub Actions CI

### PR DESCRIPTION
**Pull request by LordAro**
Saw you updating the travis CI infrastructure. Can't have that. Use something modern instead.

Basic structure largely copied from OTTD. Compiles for:

* Linux (ubuntu 20.04) with gcc & clang
* macOS
* Windows (x86 & x64)

Notable difference in the CI to OTTD's is that Windows & macOS CI links dynamically, as I couldn't manage to get static linking to work. There are various missing libraries needed to make that work, afaict - I just applied the minimal amount of changes necessary to make it work

Closes #115

### CODE REVIEW

**Comment by LordAro** [comment]
Fine GH, be that way (it won't trigger until it's merged). Run examples:

https://github.com/LordAro/FreeRCT/actions/runs/605274263
https://github.com/LordAro/FreeRCT/actions/runs/605274266 ( scan results: https://github.com/LordAro/FreeRCT/security/code-scanning?query=ref%3Arefs%2Fpull%2F1%2Fmerge )

File: .github/workflows/ci-build.yml
Comment (Noordfrees) [inline .github/workflows/ci-build.yml:4]: Perhaps this might fix it for PRs?
```suggestion
  pull_request:
    types: [ opened, reopened, synchronize ]
```

Comment (LordAro) [reply .github/workflows/ci-build.yml:4]: No, it'll work once it's merged. It worked for testing in my fork, it's just that it needs something in master first for the "base" repo before it'll do anything (at least, that's what seems to be the case)

**Review by Noordfrees — APPROVED** [review]
LGTM :)

**Comment by Alberth289346** [comment]
It's fine to merge a PR from someone else if you feel it's good :)

**Comment by Alberth289346** [comment]
Thanks Lord Aro !

**Pull request merged**
